### PR TITLE
deploy jenkins-webhook app

### PR DESCRIPTION
**Pull request by mokainos**

### CODE REVIEW

**Review by JordanHoey96 — DISMISSED** [review]

File: apps/jenkins/base/kustomization.yaml [hunk 1]
```diff
@@ -7,6 +7,10 @@ resources:
   - ../jenkins/logging.yaml
   - ../jenkins/disk.yaml
   - ../jenkins/identity.yaml
+  - ../jenkins-webhook-relay/image-policy.yaml
+  - ../jenkins-webhook-relay/image-repo.yaml
+  - ../jenkins-webhook-relay/jenkins-webhook-relay.yaml
+
```
Comment (timja) [inline apps/jenkins/base/kustomization.yaml:13]:
```suggestion

```

File: apps/jenkins/jenkins-webhook-relay/image-policy.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+apiVersion: image.toolkit.fluxcd.io/v1beta1
+kind: ImagePolicy
+metadata:
+  name: jenkins-webhook-relay
+spec:
+  imageRepositoryRef:
+    name: jenkins-webhook-relay
```
Comment (timja) [inline apps/jenkins/jenkins-webhook-relay/image-policy.yaml:7]:
```suggestion
    name: jenkins-webhook-relay

```

File: apps/jenkins/jenkins-webhook-relay/image-repo.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+apiVersion: image.toolkit.fluxcd.io/v1beta1
+kind: ImageRepository
+metadata:
+  name: jenkins-webhook-relay
+spec:
+  image: hmctspublic.azurecr.io/jenkins/jenkins-webhook-relay
```
Comment (timja) [inline apps/jenkins/jenkins-webhook-relay/image-repo.yaml:6]:
```suggestion
  image: hmctspublic.azurecr.io/jenkins/jenkins-webhook-relay

```

File: apps/jenkins/jenkins-webhook-relay/jenkins-webhook-relay.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        kind: GitRepository
+        name: jenkins-webhook-relay
+        namespace: flux-system
+      version: '*'
+      valuesFiles:
+        - ./charts/jenkins-webhook-relay/values.yaml
+  interval: 5m
+  releaseName: jenkins-webhook-relay
+  values:
+    image: hmctspublic.azurecr.io/jenkins/jenkins-webhook-relay:prod-cba235fa-1688475879 #{"$imagepolicy": "flux-system:jenkins-webhook-relay"}
```
Comment (timja) [inline apps/jenkins/jenkins-webhook-relay/jenkins-webhook-relay.yaml:22]:
```suggestion
    image: hmctspublic.azurecr.io/jenkins/jenkins-webhook-relay:prod-cba235fa-1688475879 #{"$imagepolicy": "flux-system:jenkins-webhook-relay"}

```

File: apps/jenkins/jenkins-webhook-relay/sbox-intsvc/jenkins-webhook-relay.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        queueName: "jenkins"
+        queueLength: "5"
+    secrets:
+      SERVICE_BUS_CONNECTION_STR:
+        secretRef: jenkins-webhook-relay
+        key: servicebus-connectionstring
+        disabled: false
+    environment:
+      CONNECTION_URL: "https://build.sandbox.platform.hmcts.net/github-webhook/"
+      SERVICE_BUS_QUEUE_NAME: "jenkins"
```
Comment (timja) [inline apps/jenkins/jenkins-webhook-relay/sbox-intsvc/jenkins-webhook-relay.yaml:34]:
```suggestion
      SERVICE_BUS_QUEUE_NAME: "jenkins"

```

File: apps/jenkins/sbox-intsvc/base/kustomization.yaml [hunk 6]
```diff
@@ -3,10 +3,12 @@ kind: Kustomization
...
   - ../../jenkins/sbox-intsvc/temporary-ingress.yaml
+  - ../../jenkins-webhook-relay/sbox-intsvc/jenkins-webhook-relay.enc.yaml
 
 patchesStrategicMerge:
   - ../../jenkins/sbox-intsvc/disk.yaml
   - ../../jenkins/sbox-intsvc/identity.yaml
   - ../../jenkins/sbox-intsvc/jenkins.yaml
   - ../../jenkins/sbox-intsvc/jenkins-azure-vm-agent.yaml
   - ../../jenkins/sbox-intsvc/jenkins-controller-version.yaml
+  - ../../jenkins-webhook-relay/sbox-intsvc/jenkins-webhook-relay.yaml
```
Comment (timja) [inline apps/jenkins/sbox-intsvc/base/kustomization.yaml:14]:
```suggestion
  - ../../jenkins-webhook-relay/sbox-intsvc/jenkins-webhook-relay.yaml

```

File: apps/flux-system/base/jenkins-webhook-relay-gitrepo.yaml [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  url: ssh://github.com/hmcts/jenkins-webhook-relay
+  ref:
+    branch: master
+  secretRef:
+    name: git-credentials
+  ignore: |
+    # exclude all
+    /*
+    # include charts directory
+    !/charts/
```
Comment (timja) [inline apps/flux-system/base/jenkins-webhook-relay-gitrepo.yaml:17]:
```suggestion
    !/charts/

```

File: apps/flux-system/base/kustomization.yaml [hunk 8]
```diff
@@ -7,4 +7,5 @@ resources:
   - hmctspublic-helmrepo.yaml
   - hmcts-stable-charts-gitrepo.yaml
   - incident-response-api-gitrepo.yaml
-  - plum-recipe-receiver-gitrepo.yaml
\ No newline at end of file
+  - plum-recipe-receiver-gitrepo.yaml
+  - jenkins-webhook-relay-gitrepo.yaml
```
Comment (timja) [inline apps/flux-system/base/kustomization.yaml:11]:
```suggestion
  - jenkins-webhook-relay-gitrepo.yaml

```

File: apps/jenkins/base/kustomization.yaml [hunk 9]
```diff
@@ -7,6 +7,9 @@ resources:
   - ../jenkins/logging.yaml
   - ../jenkins/disk.yaml
   - ../jenkins/identity.yaml
+  - ../jenkins-webhook-relay/image-policy.yaml
+  - ../jenkins-webhook-relay/image-repo.yaml
+  - ../jenkins-webhook-relay/jenkins-webhook-relay.yaml
 
 patchesStrategicMerge:
   - ../jenkins/jenkins-controller-version.yaml
```
Comment (timja) [inline apps/jenkins/base/kustomization.yaml:15]:
```suggestion
  - ../jenkins/jenkins-controller-version.yaml

```

File: apps/jenkins/base/kustomization.yaml [hunk 10]
```diff
@@ -7,6 +7,9 @@ resources:
   - ../jenkins/logging.yaml
   - ../jenkins/disk.yaml
   - ../jenkins/identity.yaml
+  - ../jenkins-webhook-relay/image-policy.yaml
+  - ../jenkins-webhook-relay/image-repo.yaml
```
Comment (adusumillipraveen) [inline apps/jenkins/base/kustomization.yaml:11]:
```suggestion
```

File: apps/jenkins/jenkins-webhook-relay/jenkins-webhook-relay.yaml [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+      chart: ./charts/jenkins-webhook-relay
+      interval: 1m
+      reconcileStrategy: ChartVersion
+      sourceRef:
+        kind: GitRepository
+        name: jenkins-webhook-relay
+        namespace: flux-system
+      version: '*'
+      valuesFiles:
+        - ./charts/jenkins-webhook-relay/values.yaml
```
Comment (adusumillipraveen) [inline apps/jenkins/jenkins-webhook-relay/jenkins-webhook-relay.yaml:18]: values are always picked default
```suggestion
      valuesFiles:
        - ./charts/jenkins-webhook-relay/values.yaml
```

**Review by Tyler-35 — APPROVED** [review]

**Pull request merged**